File: creacion/factory_method/java/patterns_3_factory_method_solucion/src/main/java/org/formacion/factorymethod/LavadoraFactory.java
package org.formacion.factorymethod;

// clase base que usara el cliente para crear cualquier tipo de lavadora
public abstract class LavadoraFactory {
	
	
	// proceso de construccion comun
	public Lavadora crea() {
		
		Lavadora lavadora = creaLavadora();
		
		lavadora.ponerMandos();
		lavadora.ponerTambor();
		
		return lavadora;
	}
	
	// proceso de construccion especifico de cada tipo de lavadora
	// Permite que el tipo devuelto sea un subtipo de Lavadora (y por tanto,
	//  las invocaciones al metodo crea() devuelvan tipos diferentes de lavadoras)
	protected abstract Lavadora creaLavadora();

}

File: creacion/factory_method/python/solution/src/main/interfaces/lavadoraFactory.py
#!/usr/bin/python3
# coding: utf-8

from abc import ABCMeta, abstractmethod

class LavadoraFactory(metaclass=ABCMeta):

    def crea(self):

        lavadora = self.creaLavadora()
        lavadora.ponerTambor()
        lavadora.ponerMandos()
        return lavadora

    @abstractmethod
    def creaLavadora(self):
        pass